src/kakarot/execution_context.cairo

```diff
@@ -397,4 +441,29 @@ namespace ExecutionContext {
 
         return ();
     }
+
+    // @notice Update the parent context.
+    // @dev The parent context is set to the given execution context.
+    // @param self The pointer to the execution context.
+    // @param parent_context The new parent context.
+    // @return The pointer to the updated execution context.
+    func update_parent_context(
```

Comment:
This cannot be called in a regular flow, right? Only when construction the ctx. i.e. this is an immutable field. So it;s kinda weird to have a setter for it IMO.

Reply:
actually I've used it because I wanted to reuse `kakarot.init_at_address` that does pretty much what one wants except this setting. But this make me think that we could give `init_at_address` a arg `parent_context` instead.